addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3319,6 +3417,271 @@ registry.BackgroundImage = SnippetOptionWidget.extend({
+        const defaultColors = this._getDefaultColors();
+        const shapeData = Object.assign(this._getShapeData(), newData);
+        const areColorsDefault = Object.entries(shapeData.colors).every(([colorName, colorValue]) => {
+            return colorValue === defaultColors[colorName];
+        });
+        if (areColorsDefault) {
+            delete shapeData.colors;
+        }
+        this.$target[0].dataset.oeShapeData = JSON.stringify(shapeData);
+    },
+    /**
+     * Updates/removes the shape container as needed and gives it the correct
+     * background shape.
+     *
+     * @private
+     */
+    _updateShapeContainer() {
+        const json = this.$target[0].dataset.oeShapeData;
+        let {shape, colors, flip} = json ? JSON.parse(json) : {};
+        flip = flip || [];
+        const target = this.$target[0];
+        let shapeContainer = target.querySelector(':scope > .o_we_shape');
+        if (!shape) {
+            return shapeContainer && shapeContainer.remove();
+        }
+        if (!shapeContainer) {
+            shapeContainer = document.createElement('div');
+            target.prepend(shapeContainer);
+            target.style.position = 'relative';
+            shapeContainer.nextElementSibling.style.position = 'relative';
+        }
+        shapeContainer.className = `o_we_shape o_${shape.replace('/', '_')}`;
+        ['x', 'y'].forEach(dir => {
+            shapeContainer.classList.toggle(`o_we_flip_${dir}`, flip.includes(dir));
+        });
+        $(shapeContainer).css('background-image', '');
+        if (colors) {
+            // Custom colors, overwrite shape that is set by the class
+            $(shapeContainer).css('background-image', `url("${this._getShapeSrc()}")`);
+        }
+    },
+    /**
+     * Returns the src of the shape corresponding to the current parameters.
+     *
+     * @private
+     */
+    _getShapeSrc() {
+        const {shape, colors} = this._getShapeData();
+        if (!shape) {
+            return '';
+        }
+        const queryString = Object.entries(colors)
+            .map(([colorName, colorValue]) => {
+                const encodedCol = encodeURIComponent(normalizeColor(colorValue));
+                return `${colorName}=${encodedCol}`;
+            }).join('&');
+        return `/web_editor/shape/${shape}.svg?${queryString}`;
+    },
+    /**
+     * Retrieves current shape data from the target's dataset.
+     *
+     * @private
+     */
+    _getShapeData() {
+        const defaultData = {
+            shape: '',
+            colors: this._getDefaultColors(),
+            flip: [],
+        };
+        const json = this.$target[0].dataset.oeShapeData;
+        return json ? Object.assign(defaultData, JSON.parse(json)) : defaultData;
+    },
+    /**
+     * Returns the default colors for the currently selected shape.
+     *
+     * @private
+     */
+    _getDefaultColors() {
+        const $shapeContainer = this.$target.find('>.o_we_shape')
```

Comment:
`> .` (not the first I saw, this one was the one too many who forced me to tell you ^^)